toolbox/fdc3-conformance/FDC3-2.0-Conformance-Test-Cases.md

```diff
@@ -0,0 +1,401 @@
+- `2.0-RaiseIntentFailTargetedAppResolve4`: Perform above test, but: 
+  - `fdc3.raiseIntent(‘sharedTestingIntent2’, testContextY, {appId: "<I's appId>"})`
+  - You should receive a JavaScript Error with the message `ResolveError.IntentDeliveryFailed` (as this app is configured for the intent and context pair, but adds intent listeners of the wrong type.
+  - **Note:  Test will need an extended timeout to allow for this to be returned in time by the desktop agent, which will have a vendor-defined timeout.**
+- `2.0-RaiseIntentFailTargetedAppInstanceResolve1`: Perform above test, but:
+  - First spawn an instance of App **A** and collect its `AppIdentifier` with `const appIdentifier = await fdc3.open({appId: "<A's appId>"})`.
+  - Then use `fdc3.raiseIntent(‘aTestingIntent’, testContextY, appIdentifier )` to target that instance.  
+  - You should receive a JavaScript Error with the message `ResolveError.NoAppsFound` (since A doesn't support this context type).
+- `2.0-RaiseIntentFailTargetedAppInstanceResolve2`: Perform above test, but:
+  - Use `fdc3.raiseIntent(‘aTestingIntent’, testContextY, {appId: "<A's appId>", instanceId "NonExistentInstanceId"})`.  
+  - You should receive a JavaScript Error with the message `ResolveError.TargetInstanceUnavailable `.
+
+### Raise Intent Result (void result)
+
+| App | Step           | Details                                                                                           |
+|-----|----------------|---------------------------------------------------------------------------------------------------|
+| Test   | 1. Raise          | `fdc3.raiseIntent(‘sharedTestingIntent1’, testContextY)`<br>starts app B.                       |
+| B       | 2. Receive Intent & Context | After starting up, B runs `fdc3.addIntentListener(‘sharedTestingIntent1’)` to register its listener.<br>It then receives `testContextY`, matching that sent by Test |
+| Test   | 3. IntentResolution          | The `raiseIntent` call returns an `IntentResolution` Object with an `AppIdentifer` as the `source field` with App B's `appId` and `instanceId` set.                     |
+| Test   | 4. await results          | Test should `await resolution.getResult()` on the `IntentResolution` object returned in the previous step. A promise should be returned quickly.                        |
+| B       | 5. return void          | B should return void after a short delay (e.g. 5 seconds).                        |
+| Test   | 6. receive void result          | The promise received by Test from `resolution.getResult()` should resolve to void. Confirm that the promise could be retrieved before the handler function returned and that the result was received *after* the result was returned by B, not before. I.e. confirm that `resolution.getResult() does not block until the result is returned, but rather returns a promise that can be awaited.`                      |
```

Comment:
This one's description needs improving (the await was making the test impossible to implement):

```suggestion
| Test   | 4. await results          | Test should `const resultPromise = resolution.getResult()` on the `IntentResolution` object returned in the previous step. A promise should be returned quickly.                        |
| B       | 5. return void          | B should return void after a short delay (e.g. 5 seconds).                        |
| Test   | 6. receive void result          | The promise received by Test from `resolution.getResult()` should eventually resolve to void and not be rejected. Additionally, confirm that the promise could be retrieved before the handler function returns. I.e. confirm that `resolution.getResult()` does not block until the result is returned, but rather returns a promise that can be awaited.`     |
```